test/distributed/_composable/test_checkpoint.py

```diff
@@ -184,17 +174,59 @@ def test_multi_args(self, use_reentrant: bool):
             MultiInputModel(device),
         )
         net2 = deepcopy(net1)
-        checkpoint(net2[0], use_reentrant=use_reentrant)
-        checkpoint(net2[2], use_reentrant=use_reentrant)
+        checkpoint(net2[0])
+        checkpoint(net2[2])
         x1 = torch.randn(20, 100, requires_grad=True)
         x2 = x1.clone()
         net1(x1).sum().backward()
         net2(x2).sum().backward()
         for p1, p2 in zip(net1.parameters(), net2.parameters()):
             self.assertEqual(p1.grad, p2.grad)
 
+    def test_raise_in_forward(self):
```

Comment:
nit: Maybe we could call this test `test_clears_state_on_error_in_forward()` (or similar) or somehow otherwise clarify that this is testing that `checkpoint`-specific state is cleared on an error in forward?